ostd/src/cpu/set.rs

```diff
@@ -168,26 +168,37 @@ impl AtomicCpuSet {
         Self { bits }
     }
 
-    /// Loads the value of the set.
+    /// Loads the value of the set with the given ordering.
     ///
-    /// This operation can only be done in the [`Ordering::Relaxed`] memory
-    /// order. It cannot be used to synchronize anything between CPUs.
-    pub fn load(&self) -> CpuSet {
+    /// This operation is not atomic. When racing with a [`Self::store`]
+    /// operation, this load may return a set that contains a portion of the
+    /// new value and a portion of the old value. Loads on each specific
+    /// word is atomic, and follows the specified ordering.
+    ///
+    /// Note that load with [`Ordering::Release`] is a valid operation, which
+    /// is different from the normal atomic operations. When coupled with
+    /// [`Ordering::Release`], it actually performs `fetch_or(0, Release)`.
+    pub fn load(&self, ordering: Ordering) -> CpuSet {
         let bits = self
             .bits
             .iter()
-            .map(|part| part.load(Ordering::Relaxed))
+            .map(|part| match ordering {
+                Ordering::Release => part.fetch_or(0, ordering),
+                _ => part.load(ordering),
+            })
             .collect();
         CpuSet { bits }
     }
 
-    /// Stores a new value to the set.
+    /// Stores a new value to the set with the given ordering.
     ///
-    /// This operation can only be done in the [`Ordering::Relaxed`] memory
-    /// order. It cannot be used to synchronize anything between CPUs.
-    pub fn store(&self, value: &CpuSet) {
+    /// This operation is not atomic. When racing with a [`Self::load`]
+    /// operation, that load may return a set that contains a portion of the
+    /// new value and a portion of the old value. Loads on each specific
+    /// word is atomic, and follows the specified ordering.
```

Comment:
`Loads ... is atomic`? Are you sure this is grammatically correct? I think it should be "load on each specific word is atomic".